lib/ansible/modules/cloud/azure/azure_rm_sql_database.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_sql_database
+version_added: "2.5"
+short_description: Manage an Databases.
+description:
+    - Create, update and delete an instance of Databases.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
+        required: True
+    server_name:
+        description:
+            - The name of the server.
+        required: True
+    name:
+        description:
+            - The name of the database to be operated on (updated or created).
+        required: True
+    tags:
+        description:
+            - Resource tags.
+        required: False
+    location:
+        description:
+            - Resource location.
+        required: True
+    collation:
+        description:
+            - The collation of the database. If createMode is not Default, this value is ignored.
+        required: False
+    create_mode:
+        description:
+            - "Specifies the mode of database creation.\n\nDefault: regular database creation.\n\nCopy: creates a database as a copy of an existing database.
```

Comment:
Let's not copy and paste the values from the documentation and do something like.

```
create_mode:
  description:
  - Specifies the mode of the database creation.
  - C(Default) is a regular database creation
  - C(Copy) creates a database as a copy of an existing database, I(source_database_id) must be set in this mode.
  - C(OnlineSecondary) creates a database as a readable secondary replica, I(source_database_id) must be set in this mode.
  - ... continue for each option and state what needs to be set
  default: Default